packages/x-date-pickers/src/TimeClock/TimeClock.tsx

```diff
@@ -163,14 +163,16 @@ export const TimeClock = React.forwardRef(function TimeClock<TDate extends unkno
   ref: React.Ref<HTMLDivElement>,
 ) {
   const localeText = useLocaleText<TDate>();
+  const now = useNow<TDate>();
+  const utils = useUtils<TDate>();
 
   const props = useThemeProps({
     props: inProps,
     name: 'MuiTimeClock',
   });
 
   const {
-    ampm = false,
+    ampm = utils.is12HourCycleInCurrentLocale(),
```

Comment:
I added a smart fallback to be consistent with the picker behavior

Reply:
I'd say just go with adding it in this PR.
I reckon that barely anyone was using the standalone Clock component.
But I do still think that we should include a breaking change item in migration guide as well as a changelog entry. The example should probably reference the original component name (was it `Clock`?) as well..? 🤔

Reply:
Added